File: Stack/Stack001.java
package Stack;

import java.util.EmptyStackException;

public class Stack001 {
    private Node top;

    private static class Node {
        private int data;
        private Node next;

        Node(int data) {
            this.data = data;
            this.next = null;
        }

    }

    public void push(int data) {
        Node temp = new Node(data);
        temp.next = top;
        top = temp;
    }

    public int pop() {
        if (top == null) {
            throw new EmptyStackException();
        }
        int temp = top.data;
        top = top.next;
        return temp;
    }

    public int peek() {
        if (top == null) {
            throw new EmptyStackException();
        }
        return top.data;
    }

    public static void main(String[] args) {

        Stack001 obj = new Stack001();
        obj.push(1);
        obj.push(2);
        obj.push(3);
        System.out.println(obj.peek());
        obj.pop();
        System.out.println(obj.peek());
        obj.pop();
        System.out.println(obj.peek());
        obj.pop();

    }
}

File: Stack/stack001.cpp
#include <bits/stdc++.h>
using namespace std;

// To Generlise the templeted Class
// template<typename T>;
// replace data type with this T, Now you will be able to use datas of different data types;
// Specify the type in main function
// i.e node<int> obj;
class node
{

private:
    vector<int> v;

public:
    void push(int data)
    {
        v.push_back(data);
    }
    bool isEmpty()
    {
        return v.size() == 0;
    }
    void pop()
    {
        if (!isEmpty())
        {
            v.pop_back();
        }
    }

    int peek()
    {

        return v[v.size() - 1];
    }
};

int main()
{
    node obj;
    for (int i = 0; i < 5; i++)
    {
        obj.push(i);
    }
    while (!obj.isEmpty())
    {
        cout << obj.peek() << endl;
        obj.pop();
    }
    return 0;
}